src/hypertable_cache.c

```diff
@@ -129,24 +130,51 @@ ts_hypertable_cache_invalidate_callback(void)
 
 /* Get hypertable cache entry. If the entry is not in the cache, add it. */
 TSDLLEXPORT Hypertable *
-ts_hypertable_cache_get_entry(Cache *cache, Oid relid)
+ts_hypertable_cache_get_entry(Cache *const cache, const Oid relid, const bool missing_ok)
 {
+	Hypertable *ht;
+
 	if (!OidIsValid(relid))
-		return NULL;
+		ht = NULL;
+	else
+		ht = ts_hypertable_cache_get_entry_with_table(cache, relid, NULL, NULL);
+
+	if (!missing_ok && ht == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_TS_HYPERTABLE_NOT_EXIST),
+				 errmsg("table \"%s\" is not a hypertable", get_rel_name(relid))));
```

Comment:
Note that if relid is InvalidOid, this error might be strange (e.g., `"table "NULL" is not a hypertable`

Reply:
Good point. Fixed.